conan/tools/meson/toolchain.py

```diff
@@ -193,24 +165,22 @@ def _to_meson_cppstd(self, cppstd):
                 '17': "'c++17'", 'gnu17': "'gnu++17'",
                 '20': "'c++1z'", 'gnu20': "'gnu++1z'"}.get(cppstd, "'none'")
 
-    @staticmethod
-    def _none_if_empty(value):
-        return "'%s'" % value if value.strip() else None
-
-    def _env_array(self, name):
-        import shlex
-        return shlex.split(self._build_env.get(name, ''))
-
     @property
     def _context(self):
-        project_options = []
-        for k, v in self.definitions.items():
-            project_options.append("%s = %s" % (k, self._to_meson_value(v)))
-        project_options = "\n".join(project_options)
+
+        def _to_meson_value(value):
+            # https://mesonbuild.com/Machine-files.html#data-types
+            if isinstance(value, str):
+                return "'%s'" % value
+            elif isinstance(value, bool):
+                return 'true' if value else "false"
+            elif isinstance(value, list):
```

Comment:
it's not the same as the previous implementation, it won't work for `array`, `tuple`, `dict`, only `list`, before it used to check for `Iterable` here

Reply:
Arguments are internal to the class, and they are all lists. Better simplify it (also removing the six dependency, as it is removed in 2.0, and merging this to develop2 will be easier.